core/txpool/policies/superchain_messaging.go

```diff
@@ -0,0 +1,102 @@
+package policies
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type messageSafetyLabel int
+
+const (
+	invalid messageSafetyLabel = iota
+	safe
+	finalized
+)
+
+var (
+	inboxExecuteMessageSignature = "executeMessage(address,bytes,(address,uint256,uint256,uint256,uint256))"
+	inboxExecuteMessageBytes4    = crypto.Keccak256([]byte(inboxExecuteMessageSignature))[:4]
+)
+
+type messageIdentifier struct {
+	Origin      common.Address
+	BlockNumber *big.Int
+	LogIndex    uint64
+	Timestamp   uint64
+	ChainId     *big.Int
+}
+
+// Manually parse the tx data according to the function signature. [TODO] introduce a soliabi
+// utility OR pass the entire calldata to the backend to centralize where deser logic happens
+// with minimal validation here (i.e min tx data byte length)
+func unpackInboxExecutionMessageTxData(txData []byte) (*messageIdentifier, []byte, error) {
```

Comment:
@tynes also suggested passing the calldata over the network so that the backend would be the only place we'd need to deser this data.

Might be able to get away with just checking the function signature and a minimum byte length for the arguments rather than parsing. wdyt?

Reply:
nit on function naming, its very verbose

We probably want something that abi decodes all of the calldata and returns a 3 tuple of `target`, `log` (as bytes) and `identifier`

Reply:
What is we just pass all of the data through to the backend and then we can keep more logic outside of geth?